[Github] Upgrading ISSUE_TEMPLATE to display 4 alternatives: Bug, Regression, Question or Feature Request

Sajjon opened the pull request:
Upgrading ISSUE_TEMPLATE from single markdown to 4 option selection: Bug, Regression, Question, Feature Request

### Motivation:

Having this template where when user presses `NEW ISSUE` button she has to think about what she's doing might lead to better issues submitted. 

### Modifications:

1. Replaced `ISSUE_TEMPLATE.md` with directory containing four different issues types.
2. Added new script `./scripts/environment.sh` which puts to together a summary of users environment and copies to pasteboard (asking first).


### Result:

#### New-Issue alternatives
![new_issue_options](https://user-images.githubusercontent.com/864410/73887452-dfb36d00-486b-11ea-8975-363bfdf85c77.png)

#### Bug report
With having pasted output generated from `./scripts/environment.sh`
![bug_report](https://user-images.githubusercontent.com/864410/73887500-f8bc1e00-486b-11ea-8fa0-fe45593a9e4d.png)

#### Feature request
Kind of like the same as before
![feature_request](https://user-images.githubusercontent.com/864410/73887602-2903bc80-486c-11ea-9e33-5b241b458d08.png)

Sajjon commented:
@Lukasa I've updated the script a bit, to check if we are on macOS, if so detect which version and Xcode info (might be useful, right?). The script looks like this now:
# Script
```bash
read -p "This will replace your current pasteboard. Continue? [y/n]" -n 1 -r
echo    # (optional) move to a new line
if [[ $REPLY =~ ^[Yy]$ ]]
then
	swiftversion=$(swift --version) &&
	unix_version_name=$(uname -a) &&
	i=""
	i="${i}Swift version: ${swiftversion}\n"
	i="${i}Unix version: ${unix_version_name}\n"

	# Check if OS is macOS, if so retrieve which version and Xcode version.
	if [[ "$OSTYPE" == "darwin"* ]]; then
		macos=$(defaults read loginwindow SystemVersionStampAsString | cat -) &&
		xcodebuild_version=$(/usr/bin/xcodebuild -version | grep Xcode) &&
		xcodebuild_build=$(/usr/bin/xcodebuild -version | grep Build) &&
		xcodeselectpath=$(xcode-select -p | cat -) &&
	
		i="${i}macOS version: ${macos}\n"
		i="${i}Xcode-select path: '${xcodeselectpath}\n"
		i="${i}Xcode: ${xcodebuild_version} (${xcodebuild_build})\n"
	fi

	echo "${i}" | pbcopy
	echo "Your pasteboard now contains debug info, paste it on Github"
fi
```

# Output
```
Swift version: Apple Swift version 5.1.3 (swiftlang-1100.0.282.1 clang-1100.0.33.15)
Target: x86_64-apple-darwin19.3.0
Unix version: Darwin Sajjode.local 19.3.0 Darwin Kernel Version 19.3.0: Thu Jan  9 20:58:23 PST 2020; root:xnu-6153.81.5~1/RELEASE_X86_64 x86_64
macOS version: 10.15.3
Xcode-select path: '/Applications/Xcode.app/Contents/Developer
Xcode: Xcode 11.3.1 (Build version 11C505)
```
# Issue template
Do you want me to change anything in the markdown templates? Or are they good?

Lukasa commented on scripts/environment.sh:
```diff
@@ -1,16 +1,40 @@
+##
+## This source file is part of the SwiftCrypto open source project
+##
+## Copyright (c) 2019 Apple Inc. and the SwiftCrypto project authors
```
This comment should be updated to be 2020.

Sajjon replied:
fixed, also fixed in script I added yesterday `scripts/generate_boilerplate_files_with_gyb.sh`

I also updated reference to `CONTRIBUTORS.txt`, which is now `CONTRIBUTORS.md`

Sajjon commented:
@Lukasa I fixed the comment in license (year + reference to `CONTRIBUTORS.txt`). I also fixed newline issue in `./script/environment.sh`, by adding the magic `-e` to echo. This is what the script produces:

```
Swift version: Apple Swift version 5.1.3 (swiftlang-1100.0.282.1 clang-1100.0.33.15)
Target: x86_64-apple-darwin19.3.0
Unix version: Darwin Sajjode.local 19.3.0 Darwin Kernel Version 19.3.0: Thu Jan  9 20:58:23 PST 2020
 root:xnu-6153.81.5~1/RELEASE_X86_64 x86_64

macOS version: 10.15.3
Xcode-select path: '/Applications/Xcode.app/Contents/Developer
Xcode: Xcode 11.3.1 (Build version 11C505)

```

Lukasa commented:
@swift-server-bot test this please

Lukasa commented:
It may be worth doing the header comment fixup in a separate PR just to keep the scope clear.

Sajjon commented:
@Lukasa True, good idea, I will split them into two separate PRs. I've updated this to only include Github issue template + new script `environments.sh`.

So ready to be merged I think?

Lukasa commented:
@swift-server-bot test this please

Sajjon commented:
For future reference: Regarding discussions of changes of file headers, see #24

Lukasa commented:
Ah, we need to update the sanity script. Find the line that does `sed -e 's/2018-2019/YEARS/' -e 's/2019/YEARS/'` and change it to `sed -e 's/2018-2019/YEARS/' -e 's/2019/YEARS/' -e 's/2020/YEARS/'`

Sajjon commented:
@Lukasa like so? https://github.com/apple/swift-crypto/pull/21/commits/be6e8e9a41a01d2c9c9420d425e3f2852a72aa76

Lukasa commented:
Looks right

Lukasa commented:
@swift-server-bot test this please

Lukasa commented on scripts/environment.sh:
```diff
@@ -0,0 +1,40 @@
+## Licensed under Apache License v2.0
+##
+## See LICENSE.txt for license information
+## See CONTRIBUTORS.md for the list of SwiftCrypto project authors
```
This needs to be `CONTRIBUTORS.txt` temporarily to get through CI. If we land this first, we can update #24 to change this.

Sajjon replied:
haha right, got it, I was unsure about that at first, thanks for pointing it out, updating now.

Sajjon commented:
I might be too liberal about my use of `git push -f`.... but did not wanna spam git history with trivial fixes. I tend to like `git commit -amend`.  So, it ought to be... amended now :D

Lukasa commented:
@swift-server-bot test this please

Lukasa commented:
🎉
